### DISCUSSION TITLE
`pipe-to` not grabbing selection as arguments

### DISCUSSION THREAD

**Discussion by jerabaul29** [helix-editor/helix#11305] [General]
### Summary

I wonder if `pipe-to` does not grab the selections as arguments as expected - or otherwise I do not understand correctly the documentation about it, and it would be helpful to provide more details about how it should be used :) .

### Reproduction Steps

I tried this:

1. `hx`
2. write the world "hello" on a line (without the "")
3. select the whole line with `x`
4. type `:pipe-to touch`

I expected this to happen:

5. the command `touch hello` is executed and the empty file `hello` gets created

Instead, this happened:

5. the editor displays an error message: `Shell error: touch: missing file operandTry 'touch --help' for more information.`

I have also tried a number of variations around this (putting spaces, " , etc)

### Helix log

```
2024-07-24T17:53:10.031 helix_term::commands [ERROR] Shell error: touch: missing file operand
Try 'touch --help' for more information.
```

### Platform

Linux, Ubunt 22.04

### Terminal Emulator

GNOME Terminal 3.44.0 using VTE 0.68.0 +BIDI +GNUTLS +ICU +SYSTEMD

### Installation Method

app image from latest release

### Helix Version

24.7

**Comment by pascalkuthe** [comment]
pipe-to (and pipe) pipe arguments to stdin (like `echo "<selection contents>" | <command>`) and do not specify them as an argument. This aligns with the name. You can use `<c-r>.` to insert the selection contents as an argument.

**Reply by chtenb** [reply]
How would this be done with multiple selections?

**Comment by mweihs** [comment]
`:pipe-to xargs touch`

Also works with multiple selections.
